src/trajectory.jl

```diff
@@ -246,13 +254,29 @@ function randcat(rng, zs::AbstractVector{<:PhasePoint}, unnorm_ℓP::AbstractMat
 end
 
 function samplecand(rng, τ::StaticTrajectory{MultinomialTS}, h, z)
-    zs = step(τ.integrator, h, z, τ.n_steps; res=[z for _ in 1:abs(τ.n_steps)])
+    n_steps = abs(τ.n_steps)
+    n_fwd = rand(0:n_steps) # FIXME: deal with multi-chain generically
+    zs_fwd = step(τ.integrator, h, z, n_fwd; fwd=true, res=[z for _ in 1:n_fwd])
+    n_bwd = n_steps - n_fwd
+    zs_bwd = step(τ.integrator, h, z, n_bwd; fwd=false, res=[z for _ in 1:n_bwd])
+    zs = vcat(reverse(zs_bwd)..., z, zs_fwd...)
     ℓws = -energy.(zs)
     if eltype(ℓws) <: AbstractVector
-        ℓws = hcat(ℓws...)
+        ℓws = cat(ℓws...; dims=2)
     end
     unnorm_ℓprob = ℓws
```

Comment:
Consider removing unnecessary intermediate variable `ℓws` and use `unnorm_ℓprob` (or unnorm_ℓp for consistency) in all places.